### PR TITLE
[PM-2852] Final merge from Key Migration branch to TDE Feature Branch

### PR DESCRIPTION
**Pull request by jlf0dev**
## Type of change

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Merges the final changes from the key migration to the TDE feature branch. See #5498 for background on key migration.

Most of these changes are refactors or bug fixes. There will be merge conflicts needed resolving, but the goal here is to verify these changes get reviewed before getting lost in our feature branch

## Code changes

- **file.ext:** Description of what was changed and why

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: libs/common/src/platform/abstractions/state.service.ts [hunk 1]
```diff
@@ -124,22 +124,22 @@ export abstract class StateService<T extends Account = Account> {
    * Gets the user key encrypted by the Pin key.
    * Used when Master Password on Reset is disabled
```
Comment (JaredSnider-Bitwarden) [inline libs/common/src/platform/abstractions/state.service.ts:125]: ❓ : is  "Reset" supposed to be Restart?

Comment (JaredSnider-Bitwarden) [reply libs/common/src/platform/abstractions/state.service.ts:125]: https://bitwarden.atlassian.net/wiki/spaces/~972964367/pages/512163855/TDE+Key+Migration#PIN-Key

Comment (JaredSnider-Bitwarden) [reply libs/common/src/platform/abstractions/state.service.ts:125]: Actually, can we update that to be the full name of the setting: "Used when Lock with Master Password on restart is disabled"
image

Comment (JaredSnider-Bitwarden) [reply libs/common/src/platform/abstractions/state.service.ts:125]: This applies to several other places below as well.

Comment (jlf0dev) [reply libs/common/src/platform/abstractions/state.service.ts]: Good point: [e54371c](https://github.com/bitwarden/clients/pull/5977/commits/e54371cc8b2665a1b40cb0c7c637f77240687462)

Comment (JaredSnider-Bitwarden) [reply libs/common/src/platform/abstractions/state.service.ts]: @jlf0dev, Ty!

**Review by shane-melton — APPROVED** [review]
The code change for AC looks good to go!

**Review by justindbaur — APPROVED** [review]
Good from Platform!

**Review by differsthecat — APPROVED** [review]
Looks good from vault!

**Pull request merged**